crates/ty_python_semantic/src/types/generics.rs

```diff
@@ -604,16 +604,10 @@ fn has_relation_in_invariant_position<'db>(
         ),
         // Subtyping between invariant type parameters without a top/bottom materialization involved
         // is equivalence
-        (None, None, TypeRelation::Subtyping) => derived_type.when_equivalent_to(db, *base_type),
-        (None, None, TypeRelation::Assignability) => derived_type
-            .has_relation_to_impl(db, *base_type, TypeRelation::Assignability, visitor)
+        (None, None, relation) => derived_type
+            .has_relation_to_impl(db, *base_type, relation, visitor)
             .and(db, || {
-                base_type.has_relation_to_impl(
-                    db,
-                    *derived_type,
-                    TypeRelation::Assignability,
-                    visitor,
-                )
+                base_type.has_relation_to_impl(db, *derived_type, relation, visitor)
```

Comment:
This might deserve a comment why we can't use `when_equivalent_to` here. (I was confused at first, at least.) My understanding is that it's because `when_equivalent_to` specifically checks _gradual_ equivalence, which won't give the right answer for `TypeRelation::Subtyping`. (If that's right, it suggests that the old code was wrong for the `Subtyping` case?)

Reply:
(Or maybe based on your Discord comment it's because it's faster? That would also deserve a comment)

Reply:
Heh, no, the motivation was correctness and the speed improvement was a pleasant surprise!!